Src/DependencyCollector/Shared/Implementation/HttpDiagnosticSourceListener.cs

```diff
@@ -2,31 +2,29 @@
 namespace Microsoft.ApplicationInsights.DependencyCollector.Implementation
 {
     using System;
-    using System.Collections.Concurrent;
     using System.Collections.Generic;
-    using System.Diagnostics;
-    using System.Globalization;
-    using System.Linq;
     using System.Net;
-    using Microsoft.ApplicationInsights.Common;
-    using Microsoft.ApplicationInsights.DataContracts;
-    using Microsoft.ApplicationInsights.Extensibility;
-    using Microsoft.ApplicationInsights.Extensibility.Implementation;
-    using Microsoft.ApplicationInsights.Web.Implementation;
+    using System.Reflection;
 
     /// <summary>
     /// Diagnostic listener implementation that listens for Http DiagnosticSource to see all outgoing HTTP dependency requests.
     /// </summary>
     internal class HttpDiagnosticSourceListener : IObserver<KeyValuePair<string, object>>, IDisposable
     {
         private readonly FrameworkHttpProcessing httpProcessingFramework;
-        private HttpDiagnosticSourceSubscriber subscribeHelper;
+        private readonly HttpDiagnosticSourceSubscriber subscribeHelper;
+        private readonly PropertyFetcher requestFetcherRequestEvent;
+        private readonly PropertyFetcher requestFetcherResponseEvent;
+        private readonly PropertyFetcher responseFetcher;
         private bool disposed = false;
 
         internal HttpDiagnosticSourceListener(FrameworkHttpProcessing httpProcessing)
         {
             this.httpProcessingFramework = httpProcessing;
             this.subscribeHelper = new HttpDiagnosticSourceSubscriber(this);
+            this.requestFetcherRequestEvent = new PropertyFetcher("Request");
+            this.requestFetcherResponseEvent = new PropertyFetcher("Request");
```

Comment:
There are 2 events:
1. For Start request, there is only Request in the payload
2. For Get Response: there are Request and Response in the payload.

In order to make reflection fast, we use the fancy efficient fetcher that only slow for the first time property is fetched from the anonymous object payload.
However those fetchers are different for given 2 payloads.

That's why we have 2 fetchers for 2 events